_posts/2018-10-21-nabla-k8s.md

```diff
@@ -0,0 +1,273 @@
+
+### ADD THE FOLLOWING CONFIGURATION ###
+[plugins]
+  [plugins.cri.containerd]
+  [plugins.cri.containerd.untrusted_workload_runtime]
+  runtime_type = "io.containerd.runtime.v1.linux"
+  runtime_engine = "/usr/local/bin/runnc"
+```
+
+We then run the containerd service in preparation for setting up kubernetes (Note: This may require running with `root` or `sudo`):
+
+```
+root@lumjjb-ThinkPad-P50:/home/lumjjb# /usr/local/bin/containerd --config /etc/containerd/config.toml
+
+INFO[2018-10-22T21:40:27.299644455-04:00] starting containerd                           revision=de4bb2ddfbb6b2b6a112b3478a935ca74dd7b796 version=v1.2.0-rc.0-44-gde4bb2d
+DEBU[2018-10-22T21:40:27.299684880-04:00] changing OOM score to -999
+INFO[2018-10-22T21:40:27.299892674-04:00] loading plugin "io.containerd.content.v1.content"...  type=io.containerd.content.v1
+INFO[2018-10-22T21:40:27.299909914-04:00] loading plugin "io.containerd.snapshotter.v1.btrfs"...  type=io.containerd.snapshotter.v1
+<<< TRUNCATED >>>
+```
+
+Note that this process runs as a daemon, so we will need to keep this running for
+the rest of the setup. Optionally, you may set it up as a [systemd service](https://manpages.debian.org/stretch/systemd/systemctl.1.en.html).
+
+## Setting up a CNI config
+
+Note: if you already have CNI configured, you may skip this step.
+
+The [Container Network Interface (CNI)](https://github.com/containernetworking/cni) is the interface that kubernetes uses for networking. A configuration is required for our setup using `hack/local-up-cluster.sh`. For simplicity, we use a simple bridge CNI plugin. 
+
+This can be done by adding the following configuration in the following two files (`/etc/cni/net.d/10-mynet.conflist`, `/etc/cni/net.d/99-loopback.conf`). This configuration will create a bridge with name `cni0` and use it to connect the pods created in the kubernetes cluster.
+
+```
+lumjjb@lumjjb-ThinkPad-P50:~$ cat /etc/cni/net.d/10-mynet.conflist
+
+{
+    "cniVersion": "0.3.1",
+    "name": "mynet",
+    "plugins": [
+        {
+            "type": "bridge",
+            "bridge": "cni0",
+            "isGateway": true,
+            "ipMasq": true,
+            "ipam": {
+                "type": "host-local",
+                "subnet": "10.30.0.0/16",
+                "routes": [
+                    { "dst": "0.0.0.0/0"   }
+                ]
+            }
+        },
+        {
+            "type": "portmap",
+            "capabilities": {"portMappings": true},
+            "snat": true
+        }
+    ]
+}
+
+lumjjb@lumjjb-ThinkPad-P50:~$ cat /etc/cni/net.d/99-loopback.conf
+
+{
+    "cniVersion": "0.3.1",
+    "type": "loopback"
+}
+```
+
+## Setup a local kubernetes cluster with containerd
+
+Finally, to tie everything in together, we will deploy kubernetes!
+
+The only changes to a deployment of a kubernetes cluster is that we need to explicitly tell the kubelet to use the containerd endpoint that we've created. For each worker, this can be done by 
```

Comment:
Wait, all the steps above assume that kubernetes is already installed?

Reply:
So I wanted to say what is needed in a generic cluster scenario, then go ahead to talk about what to do for a local start of a kubernetes cluster. Let me re-organize this for more clarity.